### DISCUSSION TITLE
systray: first icon in tray is black or invisible

### DISCUSSION THREAD

**Discussion by bakkeby** [bakkeby/dwm-flexipatch#136] [Known Issues]
This issue can be reproduced by having an empty systray and running this command:
```
$ nm-applet --indicator
```
Compared to running this command:
```
$ nm-applet
```

The reason for this issue is that depending on the height of the bar (e.g. 22) the first program launched may use the appindicator protocol rather than the older xembed protocol and this causes the icon to be black or invisible.

The systray patches for dwm only supports the xembed protocol and so far there is no proper fix for this issue.

#### Workaround

Adjust the bar height by either:
   - changing the font size or
   - use the [bar height](https://dwm.suckless.org/patches/bar_height/) patch or
   - use the [statuspadding](https://dwm.suckless.org/patches/statuspadding/) patch

This is ref. https://www.reddit.com/r/suckless/comments/mwb2lv/dwmsystray62diff_patch_sometimes_no_icons_in_tray/

**Comment by explosion-mental** [comment]
I stumble across this as well. Some other workarounds I found:
- restarting dwm
- spawning another 'bug'ed systray, like `blueman-applet` or element one, etc. Seems to refresh them..
